core/libp2p/multi/multibase/codecs/base58.hpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_BASE58_HPP
+#define KAGOME_BASE58_HPP
+
+#include <optional>
+
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
+/**
+ * Encode/decode to/from base58 format
+ * Implementation is taken from
+ * https://github.com/bitcoin/bitcoin/blob/master/src/base58.h
+ */
+namespace libp2p::multi::detail {
+  /**
+   * Encode bytes to base58 string
+   * @param bytes to be encoded
+   * @return encoded string
+   */
+  std::string encodeBase58(const kagome::common::Buffer &bytes);
```

Comment:
is this bitcoin variant?

Reply:
Yes, as it is written in the header comment, implementation is taken from the Bitcoin repo